src/driver/mongodb.rs

```diff
@@ -0,0 +1,261 @@
+            match query {
+              // TODO: Make this a range error when implementing selection by
+              // range.
+              Query::Value => {
+                let mut cursor = try!(collection.find(None, None));
+                let mut values = Vec::new();
+                if let Some(Err(error)) = cursor.find(|entry| match entry {
+                  &Ok(ref document) => { values.push(Value::from(document.clone())); false },
+                  &Err(_) => true
+                }) {
+                  return Err(Error::from(error));
+                } else {
+                  object.insert(coll_name, Value::Array(values));
+                }
+              },
+              // TODO: When implementing collections consider not using the
+              // MongoDB `_id` property as the key.
+              Query::Object(_) => {
+                
+              }
+            }
+          // } else {
+          //   return Err(Error::new(
+          //     ErrorCode::Forbidden,
+          //     "Can‘t query a selection of MongoDB collections.",
+          //     Some("Query specific collections instead of a range of collections.")
+          //   );
+          // }
+        }
+        Ok(Value::Object(object))
+      }
+    }
+  }
+  
+  fn patch(&self, _: Vec<Patch>) -> Result<Value, Error> {
+    Err(Error::unimplemented("Patching not implemented for MongoDB driver."))
+  }
+}
+
+impl From<MongoDBError> for Error {
+  fn from(error: MongoDBError) -> Self {
+    Error::internal(error.description())
+  }
+}
+
+impl From<Document> for Value {
+  fn from(document: Document) -> Self {
+    let mut map = LinearMap::new();
+    for key in &document.keys {
+      if let Some(value) = document.get(key) {
+        map.insert(key.clone(), Value::from(value.clone()));
+      }
+    }
+    Value::Object(map)
+  }
+}
+
+impl From<Bson> for Value {
+  fn from(bson: Bson) -> Self {
+    match bson {
+      Bson::FloatingPoint(value) => Value::F64(f64::from(value)),
+      Bson::String(value) => Value::String(value),
+      Bson::Array(values) => Value::Array(values.into_iter().map(|v| Value::from(v)).collect()),
+      Bson::Document(document) => Value::from(document),
+      Bson::Boolean(value) => Value::Boolean(value),
+      Bson::Null => Value::Null,
+      Bson::RegExp(value, _) => Value::String(value),
+      Bson::JavaScriptCode(value) => Value::String(value),
+      Bson::JavaScriptCodeWithScope(value, _) => Value::String(value),
+      Bson::I32(value) => Value::I64(i64::from(value)),
+      Bson::I64(value) => Value::I64(i64::from(value)),
+      Bson::TimeStamp(value) => Value::I64(i64::from(value)),
+      // TODO: Actual transformation of binary type.
+      Bson::Binary(_, _) => Value::Null,
+      Bson::ObjectId(object_id) => Value::String(object_id.to_string()),
+      Bson::UtcDatetime(time) => Value::String(time.to_rfc3339())
+    }
+  }
+}
+
+// fn query_to_projection(query: Query) -> Bson {
```

Comment:
Why the commented out code?
This is why git exists, just push it as a commit and revert it if you ever need it again.